databases_preview.go

```diff
@@ -0,0 +1,125 @@
+package scalingo
+
+import (
+	"context"
+	stderrors "errors"
+
+	"github.com/Scalingo/go-utils/errors/v2"
+)
+
+const databasesPreviewResource = "databases"
+
+var ErrDatabaseNotFound = stderrors.New("database not found")
+
+type PreviewAPI interface {
+	DatabasesPreviewService
+}
+
+type DatabasesPreviewService interface {
+	DatabaseCreate(ctx context.Context, params DatabaseCreateParams) (DedicatedDatabase, error)
+	DatabasesList(ctx context.Context) ([]DedicatedDatabase, error)
+	DatabaseShow(ctx context.Context, appID string) (DedicatedDatabaseShowResponse, error)
```

Comment:
This should return the generic model no?
```suggestion
	DatabaseShow(ctx context.Context, appID string) (DedicatedDatabase, error)
```

Reply:
Got it thanks. Shouldn't we have only one struct, ideally, for DBs?